test/test_spectral_ops.py

```diff
@@ -332,6 +332,32 @@ def test_fft_backward(self, device, dtype):
                 args = args[1:]
                 self._fft_grad_check_helper(fname, input, args)
 
+    @skipCPUIfNoMkl
+    @skipCUDAIfRocm
+    @onlyOnCPUAndCUDA
+    def test_fft_invalid_out_types(self, device):
+
+        complex_fft_funcs = [torch.fft.fft, torch.fft.ifft, torch.fft.fftn, torch.fft.ifftn,
+                             torch.fft.rfft, torch.fft.rfftn, torch.fft.ihfft]
+        real_fft_funcs = [torch.fft.irfft, torch.fft.irfftn, torch.fft.hfft]
+        fft_funcs = complex_fft_funcs + real_fft_funcs
+
+        # Test errors on invalid out dtypes
+        x = torch.rand(10, device=device, dtype=torch.float32)
+        for out_dtype in (torch.int16, torch.float32, torch.complex64):
```

Comment:
Code golf: this could just be an iterable of pairs like (out, dtype, funcs to test):

```
for out_dtype, funcs in ((torch.int16, fft_funcs), ...):
```